### ISSUE TITLE
We need your help to help!

### ISSUE THREAD

**Issue by thomiel** [jsuarezruiz/maui-linux#34]
### Description

I was wondering if this project shouldn't actively invest more in community building and make it easier for for non-MVP to contribute.

I think, the first step to more open up to the Linux community here could be to document install/build steps for building maui-linux on a native Linux system. How about a tutorial showing setup of a dev environment with Raider or VS Code? Maybe we could organize a workshop or something.

This project has the potential to really prove all the folks wrong who are whining about Microsoft not loving Linux (desktop) hard enough! It would be nice if this would be more of a community crystallization point rather than a mere pull request processor.

### Steps to Reproduce

n/a

### Version with bug

Unknown/Other (please specify)

### Last version that worked well

Unknown/Other

### Affected platforms

Other (Tizen, Linux, etc. not supported by Microsoft directly)

### Affected platform versions

Ubuntu 20.04 LTS

**Comment by ImJakeLu** [comment]
agree

**Comment by BernhardNinaus** [comment]
That would be really nice!

I just have to much free time and really want to do something for the community!

Tank you so far!

**Comment by jsuarezruiz** [comment]
> I have created a maui-linux room on Matrix: https://matrix.to/#/#maui-linux:matrix.org -- I hope this is acceptable for most of you for the start. If you think another platform would be more suitable, go ahead. The more the merrier! (But I personally think it should be an open platform.) At some point we could do a pull request with active channels in the README.md to make them somewhat _official_.
> 
> I will crosspost this on Issue #3 as well.

Is fine, I can ask if we can have a channel in the DotNet Discord.

**Comment by jsuarezruiz** [comment]
> @jsuarezruiz What do you think about reaching out to the .NET Foundation to back the project administratively?

Yes, I certainly think it would help. I'll try to move it in that direction.

**Comment by mcopjan** [comment]
Can anyone provide some information, how to actually use this repo? E.g. is it possible to compile BlazorWpfApp from this repo to target Ubuntu and then run this on Ubuntu when .net 6 is installed there?

**Comment by thomiel** [comment]
Unfortunately, the repo isn't currently in a good shape to be used. One cause that I suspect is the somewhat unclear release state of GTK 3. This causes problems under different conditions, e.g. the GtkSharp version referenced in branch main-linux can't be found on nuget and there is no proper Gtk installer for Windows available. And I think the repo badly needs some upstream love. And this is _can be used_ in terms of _it's building and ready to be developed further_ -- and not in the sense of _can be used in production_. After all, the MAUI without GTK wasn't finally release and is still preview.

The [the GTK backend for Xamarin.Forms](https://github.com/jsuarezruiz/forms-gtk-progress) relies on GTK 2 and is much more usable for production imho (although I stumbled upon problems with the latest Debian where two graphic libraries were no longer supplied by the packet manager.) But of course, this won't help you as you obviously want to do something using BlazorWebView.

**Comment by tripleo1** [comment]
> One cause that I suspect is the somewhat unclear release state of GTK 3.

I don't know if you mean GtkSharp 3.xx, because GTK (on Linux) is up to version 4 (although version 3 is very much alive). 
Also, I suspect many people that are _here_ in the first place are not worried too much about GTK in any form running on Windows or Mac -- as far as I know, they both have support from MS.

I just want to port some C# apps that I find on GitHub to my Linux system without rewriting them completely in another language.

BTW, the command line is as important as IDEs... (but VS Code is acceptable too).

**Comment by thomiel** [comment]
> > One cause that I suspect is the somewhat unclear release state of GTK 3.
> 
> I don't know if you mean GtkSharp 3.xx, because GTK (on Linux) is up to version 4 (although version 3 is very much alive).

You are right. It's GTK#. See [here](https://en.wikipedia.org/wiki/GTK#Gtk#):

> As of September 2020, Gtk# support for Gtk3 remains in the preview phase and forked projects, such as [GtkSharp](https://github.com/GtkSharp/GtkSharp), have been founded to provide full Gtk3 support for C# and other CLI languages. The lack of a released version of Gtk# with support for Gtk3 was cited as a reason to remove the [Banshee](https://en.wikipedia.org/wiki/Banshee_(media_player)) media player in Ubuntu 12.04

**Comment by microhobby** [comment]
Any news about this? The .NET Foundation back and the discord community?

**Comment by thomiel** [comment]
There was some activity via the Matrix room, a .NET 6 workload for GtkSharp and adaptions in Microsoft.Maui.Graphics regarding Gtk. (Microsoft.Maui.Graphics seems to have moved into the Microsoft.Maui repo though.)

Seems that there is a tendency within Microsoft to proceed wit Gtk integration in Maui for .NET 7, if I interpret Discord murmur right.

**Comment by microhobby** [comment]
@thomiel wow, this is awesome news! 
@jsuarezruiz could you confirm something in advance?

**Comment by jsuarezruiz** [comment]
> @thomiel wow, this is awesome news! @jsuarezruiz could you confirm something in advance?

There's definitely internal conversations about Linux, but I can't share anything right now.

**Comment by lytico** [comment]
i started merging recent maui: https://github.com/jsuarezruiz/maui-linux/pull/37

**Comment by kdotdk** [comment]
> There's definitely internal conversations about Linux, but I can't share anything right now.
I read in another posting that Linux was a marginal (Ed: Desktop) OS, hence it was not interesting for Microsoft to supply MAUI support. But if the MAUI Linux disk/ram footprint is not too big - the marked for UI is very large on embedded Linux systems.

**Comment by pha3z** [comment]
I don't have a lot of optimism about this project succeeding.

But honestly it got me to wondering why we can't user a different approach.  If you're deploying a Blazor Hybrid app across all other platforms with MAUI,  you could just deploy your Blazor app in Server mode on Linux and then use any run-of-the-mill browser or wrapped-webview to connect to the app.  99% of Linux users are going to be comfortable with having a web server listening on localhost and using that as the app host.

Anyone else see a reason why this wouldn't be a viable solution to get Blazor on Linux?

**Comment by EricOnGit** [comment]
> I don't have a lot of optimism about this project succeeding.
> 
> But honestly it got me to wondering why we can't user a different approach. If you're deploying a Blazor Hybrid app across all other platforms with MAUI, you could just deploy your Blazor app in Server mode on Linux and then use any run-of-the-mill browser or wrapped-webview to connect to the app. 99% of Linux users are going to be comfortable with having a web server listening on localhost and using that as the app host.
> 
> Anyone else see a reason why this wouldn't be a viable solution to get Blazor on Linux?

I can see the hardware integration as less viable in this scenario. Accessing usb, bluetooth, wifi are nice tricks that MAUI brings to the table that are much less obvious when running in a browser. Sure you can move this code to the web server but it smells just writing it... 

Otherwise, I think you're right. Local Server + Browser will have to suffice, at least for now.

**Comment by seriouz** [comment]
> > @thomiel wow, this is awesome news! @jsuarezruiz could you confirm something in advance?
> 
> There's definitely internal conversations about Linux, but I can't share anything right now.

I want to run MAUI on my Steam Deck!

**Comment by TenEniunlsl** [comment]
While i'm on deep dive in the project and make some code changes i have started in parallel a  more helpfull wiki and a news blog see [maui-linux.net](https://maui-linux.net)

**Comment by Lemon73-Computing** [comment]
> While i'm on deep dive in the project and make some code changes i have started in parallel a more helpfull wiki and a news blog see [maui-linux.net](https://maui-linux.net)

That's so interesting. And thank you!

**Comment by Lemon73-Computing** [comment]
@TenEniunlsl I added the website information into MauiGtk summary!

https://github.com/MauiGtk/.github/blob/main/profile/README.md#community-%E2%80%8D%E2%80%8D

**Comment by thomiel** [comment]
> While i'm on deep dive in the project and make some code changes i have started in parallel a more helpfull wiki and a news blog see [maui-linux.net](https://maui-linux.net)

Cool! But why didn't you officially ask to join the MauiGtk org? No need for a shism that early in the project. ;)

PS: You copied a lot of stuff into your wiki but I'm not sure if it's more helpful than ours. For example it states that [HavenDV](https://github.com/HavenDV) did the Gtk Workflow and not @trungnt2910 ... or maybe I'm confusing it with the GtkSharp workflow...?

**Comment by trungnt2910** [comment]
> For example it states that [HavenDV](https://github.com/HavenDV) did the Gtk Workflow and not [@trungnt2910](https://github.com/trungnt2910) ... or maybe I'm confusing it with the GtkSharp workflow...?

There does seem to be an independent effort by [HavenDV](https://github.com/HavenDV/Gtk) for a GTK workload, but the first commit is in [2023](https://github.com/HavenDV/Gtk/commit/f259c393caa0bcc0df52a08e395852facbac424d), which is way after mine have been merged in [May 2022](https://github.com/GtkSharp/GtkSharp/pull/351).

**Comment by TenEniunlsl** [comment]
first things first no one will search for mauigtk, me as example, i have searched every couple of months in the last 3 years for updates around maui linux, four weeks ago i found mauigtk ;-) in this [issue](https://github.com/MauiGtk/maui-linux/issues/109)

Then there was'nt a easy to use tutorial to build and run it, so i decided too change that

> You copied a lot of stuff into your wiki but I'm not sure if it's more helpful than ours

Oh it is and will be, an example you can download a script to create your Development Environment or download a fully working xubuntu VM and a page about Visual Studio 2022 and WSL
then there isn't copied anything but a little bit help of ChatGPT
The next step will be integrate the old wiki in the new and complete [this pages](https://wiki.maui-linux.net/index.php/Maui_Sample_Project_State)

Also i had said i have started not it's finished ;-) but since i'm writing on it i also had implemented some new things, as example:
i changed the HavenDV workflow linked [here](https://github.com/MauiGtk)
i think it should be the right one or the mauigtk project page lies

> or maybe I'm confusing it with the GtkSharp workflow

There is also another workflow? Tell me more, witch isn't documented yet in a way to find it ;-)

**Comment by TenEniunlsl** [comment]
you can also contribute on the wiki you only need to register an account

**Comment by thomiel** [comment]
> first things first no one will search for mauigtk, me as example, i have searched every couple of months in the last 3 years for updates around maui linux, four weeks ago i found mauigtk ;-) in this [issue](https://github.com/MauiGtk/maui-linux/issues/109)

Our propaganda machine hasn't been started yet. :D

> Oh it is and will be, an example you can download a script to create your Development Environment or download a fully working xubuntu VM and a page about Visual Studio 2022 and WSL then there isn't copied anything but a little bit help of ChatGPT The next step will be integrate the old wiki in the new and complete [this pages](https://wiki.maui-linux.net/index.php/Maui_Sample_Project_State)

I want to encourage your efforts, but it would be preferable to do it in a way that would make it easier to merge things.

> Also i had said i have started not it's finished ;-) but since i'm writing on it i also had implemented some new things, as example: i changed the HavenDV workflow linked [here](https://github.com/MauiGtk) i think it should be the right one or the mauigtk project page lies

Can't find it. Please [can you tell me the line number](https://github.com/MauiGtk/.github/blob/main/profile/README.md?plain=1)?

> > or maybe I'm confusing it with the GtkSharp workflow
> 
> There is also another workflow? Tell me more, witch isn't documented yet in a way to find it ;-)

https://github.com/GtkSharp/GtkSharp/pulls?q=is%3Apr+is%3Aclosed+Workload

Your wiki also references maui-docker; see here: https://github.com/MauiGtk/maui-docker/blob/main/Dockerfile#L18

**Comment by TenEniunlsl** [comment]
> Our propaganda machine hasn't been started yet. :D

so i have build one ;-)

> Can't find it. Please can you tell me the line number

Sure this one
https://github.com/MauiGtk/.github/blob/699eff9c8fd324d9d1bc7827629ee83e8fd8b893/profile/README.md?plain=1#L19

**Comment by TenEniunlsl** [comment]
> Your wiki also references maui-docker; see here: https://github.com/MauiGtk/maui-docker/blob/main/Dockerfile#L18

And now say it >> the code is the documentation ;-)

**Comment by thomiel** [comment]
> > Your wiki also references maui-docker; see here: https://github.com/MauiGtk/maui-docker/blob/main/Dockerfile#L18
> 
> And now say it >> the code is the documentation ;-)

Not quite Ansible, but I find a Dockerfile quite instructive, sure! B)

But now I see it: @Lemon73-Computing has confused Work_flow_ and Work_load_ in https://github.com/MauiGtk/.github/commit/ac43dc99fbdeb66b5811deaa1b20e93f1d3fb517 -- it wasn't even a ChatGPT hallucination but a simple copycat error!

**Comment by Lemon73-Computing** [comment]
> But now I see it: @Lemon73-Computing has confused Work_flow_ and Work_load_ in https://github.com/MauiGtk/.github/commit/ac43dc99fbdeb66b5811deaa1b20e93f1d3fb517 -- it wasn't even a ChatGPT hallucination but a simple copycat error!

What do you mean? I think “Workload” is correct, since HavenDV uses [the word](https://github.com/HavenDV/Gtk#:~:text=About-,GTK%20workload,-and%20separate%20%60net7.0) in his repo. (but if I’m mistaken, I’ll change it to “Workflow” :eyes:)

**Comment by TenEniunlsl** [comment]
> Not quite Ansible, but I find a Dockerfile quite instructive, sure! B)

Maybe cause you are a expierenced docker user? But i don't work with docker normally i use VMs, Windows and Visual Studio 2022 with WSL to Develop and have an LPIC1 and 2 and also 15years of expierence in VB.NET and C#

Everything you said against maui-linux.net just confirmed the reasons why I started doing this

**Comment by thomiel** [comment]
> > Our propaganda machine hasn't been started yet. :D
> 
> so i have build one ;-)

Fine. Then let us use it wisely! Instead of duplicating the wiki with all the technical details that at least partially better belong to decentralized repo readmes, I'd opt for more comprehensive info. The blog is a good thing for sure. And some shiny content for folks to understand at a single glance what we are trying to do. This would be the place to showcase things, not to document them in detail.

Is there a chance we can convinve you to change the domain name from traditionalist _maui-linux.net_ to _MauiGtk.net_?

**Comment by thomiel** [comment]
> > But now I see it: [@Lemon73-Computing](https://github.com/Lemon73-Computing) has confused Work_flow_ and Work_load_ in [MauiGtk/.github@ac43dc9](https://github.com/MauiGtk/.github/commit/ac43dc99fbdeb66b5811deaa1b20e93f1d3fb517) -- it wasn't even a ChatGPT hallucination but a simple copycat error!
> 
> What do you mean? I think “Workload” is correct, since HavenDV uses [the word](https://github.com/HavenDV/Gtk#:~:text=About-,GTK%20workload,-and%20separate%20%60net7.0) in his repo. (but if I’m mistaken, I’ll change it to “Workflow” 👀)

Sorry, my bad. I definitely need more time to read through the repos...

**Comment by TenEniunlsl** [comment]
> > > Our propaganda machine hasn't been started yet. :D
> > 
> > 
> > so i have build one ;-)
> 
> Fine. Then let us use it wisely! Instead of duplicating the wiki with all the technical details that at least partially better belong to decentralized repo readmes, I'd opt for more comprehensive info. The blog is a good thing for sure. And some shiny content for folks to understand at a single glance what we are trying to do. This would be the place to showcase things, not to document them in detail.
> 
> Is there a chance we can convinve you to change the domain name from traditionalist _maui-linux.net_ to _MauiGtk.net_?

It's like selling drugs, you have to hook people first, so like I said, no one is searching for Mauigtk, every newbie will be searching for Maui Linux, then you can show them the complete world around

**Comment by thomiel** [comment]
... says the one with the unspeakable user name. 😆

I wish you the best with maui-linux.net then.

**Comment by TenEniunlsl** [comment]
> ... says the one with the unspeakable user name. 😆
> 
> I wish you the best with maui-linux.net then.

It's easy call me ten like 10
